CoreGraphFramework/src/au/gov/asd/tac/constellation/graph/StoreGraph.java

```diff
@@ -1999,10 +2000,16 @@ private void recalculateLayerVisibilities() {
      */
     private int recalculateLayerMask(final GraphElementType elementType, final int elementId) {
         int bitmask = 0;
-        if (elementType == GraphElementType.VERTEX && vertexLayerMaskAttribureId >= 0 && vertexLayerVisibilityAttributeId >= 0) {
-            bitmask = getIntValue(vertexLayerMaskAttribureId, elementId);
-        } else if (elementType == GraphElementType.TRANSACTION && transactionFilterBitmaskAttrId >= 0 && transactionLayerVisibilityAttributeId >= 0) {
-            bitmask = getIntValue(transactionFilterBitmaskAttrId, elementId);
+        if (elementType == GraphElementType.VERTEX) {
+            vertexLayerMaskAttribureId = GraphManager.getDefault().getActiveGraph().getReadableGraph().getAttribute(GraphElementType.VERTEX, LAYER_MASK_ATTRIBUTE_LABEL);
```

Comment:
Whenever you get a `ReadableGraph` or `WritableGraph` you should ensure that you release the lock when you're done with it. In this case i'd suggest something like:
```
final readableGraph = GraphManager.getDefault().getActiveGraph().getReadableGraph();
try {
    vertexLayerMaskAttributeId = VisualConcept.VertexAttrtibute.LAYER_MASK.get(readableGraph);
} finally {
    readableGraph.release();
}
```

Reply:
Also, `StoreGraph` is a `GraphWriteMethods`... so couldn't you just write `vertexLayerMaskAttributeId = VisualConcept.VertexAttrtibute.LAYER_MASK.get(this)`?

Reply:
Further to this, with context managers available as of Java 9? (8? 7?) could we update the graph API to allow:
```
try (graph.getReadableGraph()) {
    ///do stuff...
}
```